draft-packages/tooltip/KaizenDraft/Tooltip/Tooltip.tsx

```diff
@@ -145,6 +156,39 @@ const TooltipContent = ({
   ) : null
 }
 
+const isSemanticElement = (
+  elementType: string | JSXElementConstructor<any>
+): boolean => {
+  if (elementType === "div" || elementType === "span") {
+    return false
+  }
+  return true
+}
+
+/**
+ *  validates implicit or explicity semantic role require to make aria-describedby announce predictably with screen readers
+ */
+const hasSemanticRole = (element: ReactNode): boolean => {
+  if (!React.isValidElement(element)) return false
+  const { props, type } = element
+
+  if (isSemanticElement(type) || "role" in props) return true
+
+  return false
```

Comment:
What if we put all of this into `isSemanticElement` so you can just pass the element in :)